src/EnergyPlus/EPVector.hh

```diff
@@ -0,0 +1,293 @@
+//     written permission.
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef EPVector_hh_INCLUDED
+#define EPVector_hh_INCLUDED
+
+#include <numeric>
+#include <vector>
+
+
+namespace EnergyPlus {
+
+template <typename T> struct EPVector : private std::vector<T>
+{
+    using std::vector<T>::size;
+    using std::vector<T>::operator[];
+    using std::vector<T>::empty;
+    using std::vector<T>::begin;
+    using std::vector<T>::end;
+    using std::vector<T>::cbegin;
+    using std::vector<T>::cend;
+    using std::vector<T>::clear;
+    using std::vector<T>::emplace_back;
+    using std::vector<T>::push_back;
+    using const_iterator = typename std::vector<T>::const_iterator;
+
+    using value_type = T;
+    using size_type = typename std::vector<T>::size_type;
+
+    [[nodiscard]] T &operator()(std::size_t n)
+    {
+        return (*this)[n - 1];
+    }
+
+    [[nodiscard]] const T &operator()(std::size_t n) const
+    {
+        return (*this)[n - 1];
+    }
+
+    void allocate(std::size_t size)
+    {
+        m_allocated = true;
+        this->resize(size);
+        std::fill(begin(), end(), T{});
+    }
+
+    void deallocate() noexcept
+    {
+        this->clear();
+    }
+
+    [[nodiscard]] bool allocated() const noexcept
+    {
+        return m_allocated || !this->empty();
+    }
```

Comment:
But then I do see the allocated member method here.  Can you clarify @lefticus?  Mostly just so I know what to look for if we spread this further into the codebase.